lib/net/ssh/multi/session.rb

```diff
@@ -447,7 +450,7 @@ def io_select_wait(wait)
 
     def keepalive_interval
       servers = server_list.select { |s| s.options[:keepalive] }
-      servers.map { |s| s.options[:keepalive_interval] }.compact.min
+      servers.map { |s| s.options[:keepalive_interval] || DEFAULT_IO_SELECT_TIMEOUT }.compact.min
```

Comment:
My proposal intended to solve `DEFAULT_IO_SELECT_TIMEOUT` by going through [this method](https://github.com/net-ssh/net-ssh/blob/v3.1.1/lib/net/ssh/connection/keepalive.rb#L23).

`server_list` is a collection of `Net::SSH::Multi::Server` objects. If we want to avoid duplicating logic from Net::SSH, `max_select_wait_time` will have to talk to its `session` attribute, which is a `Net::SSH::Connection::Session`. Without a way to fake a session in this gem's test suite, it will be impossible to write tests.

Sorry for the back and forth here, but I'm not comfortable adding an interface between the two gems without being able to test it :confused: